### ISSUE TITLE
bug: Misleading `console.error` when rendering components using `internals` SSR

### ISSUE THREAD

**Issue by dwirz** [ionic-team/stencil#6073]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.22.3

### Current Behavior

Currently, if a component has the `formAssociated` flag set and uses `@AttachInternals()`, calling `setValidity` within the `componentDidLoad` lifecycle method results in the following `console.error` message being logged:  `NOTE: Property form was accessed on...`

This occurs because the underlying `MockElement` implementation proxies all calls to `internals` which calls `console.error`. While this behavior might be reasonable in a testing environment—where it can serve as a helpful warning—it becomes irrelevant and misleading in server-side rendering (SSR) contexts.  

In SSR, such errors are unnecessary since the `ElementInternals` API does not apply in the server environment. Addressing this could involve conditionally suppressing these errors during SSR to avoid confusion and ensure a cleaner debugging experience.

### Expected Behavior

The `console.error` shouldn't be logged if the component is rendered in a SSR enviroment.

### System Info

```shell
System: node 22.11.0
       Build: 1732216626
     Stencil: 4.22.3 🐤
  TypeScript: 5.5.4
```

### Steps to Reproduce

See Description in [dwirz/nextjs-stenciljs-example](https://github.com/dwirz/nextjs-stenciljs-example)

### Code Reproduction URL

https://github.com/dwirz/nextjs-stenciljs-example

### Additional Information

Actually it is not a real bug, but it IMO the message should not be displayed if a component is rendered SSR because it is misleading.

The message comes from [mock-doc/node.ts#L234](https://github.com/ionic-team/stencil/blob/c2e97f797f6eb7b843aa153fef2780073ec76898/src/mock-doc/node.ts#L234)

At the moment we add the following workaround check to each component which uses `internals`:
```
  componentDidLoad() {
    if (typeof window === "undefined" || window.constructor.name !== "Window") {
      return;
    }
  
    // ...
    
    this.internals.setValidity({});
  }
```

**Comment by christian-bromann** [comment]
@dwirz thanks for raising the issue. I totally agree, and raised a PR with a fix in #6075

**Comment by christian-bromann** [comment]
A fix for this was released in Stencil [`v4.23.0`](https://github.com/ionic-team/stencil/releases/tag/v4.23.0)
